slm_porosity/heat_transfer.f90

```diff
@@ -605,6 +621,16 @@ SUBROUTINE user_read_input ()
     Dphi_one = 1.0_pr/(enthalpy_L - enthalpy_S)
     interface_thickness = (enthalpy_L - enthalpy_S - fusion_heat)/capacity(1.0_pr, 0.5_pr)
 
+    IF (dim.EQ.2) THEN
+      call input_real_vector ('coord_min', coord_min, 3, 'stop')
+      call input_real_vector ('coord_max', coord_max, 3, 'stop')
+      xyzlimits(2,2) = coord_max(3)
```

Comment:
This is a bad solution, because
1. You change the global variables and can not even imagine implications.
2. Repeated `input_real`
3. In this function you should deal only with user-defined variables.

Why do not you like the current implementation?

Reply:
There was a discussion about saving XYZ XZ coordinate system notation. This helps avoid such construction as `laser_position(dim) = xyzlimits(2, dim)` in `laser_position` which leads `initial_laser_position` variable is ignored in some cases. Which is harmful for internal laser heating. 

1. It should not have any conflicts with the old code.

2. I really don't understand how I can't avoid reading two different variables with one input_real

3. I reassign the xyzlimits(:,2), with saving its dimension. What bad in this?

Reply:
I comprehend the situation. Use reassign xyzlimits long after its initialization. Possibly, it was already been used before `user_read_input`. Actually, it is true if you look through `io_3d.f90`. Anyway, it is not your variable and it is potentially harmful to change it, because the current architecture does not imply this functionality.  Weighting all pros and cons, I suggest to change Y and Z axes in our code. It will contradict to our manual, but not in the dangerous way. This solution require much more amendments, but the most elegant as I see. What do you think?

Reply:
In the way that domain could be initilized before user_read_input the changing of axis numeration seems reasonable.

Reply:
Will you implement it?